### PR TITLE
enhanced bootstrapping flow documentation

### PR DESCRIPTION
**Pull request by baronlior**
Hi,
I created the original document for internal use - but thought of contributing it back to the community.
This pull request is just to check you are interestested in this contribution, before I invest extra effort.
A concept being used is "simplified code" -> showing code, but in a simpler form that is easier to follow.

Work still needed:
* Translate two code snippets from kotlin --> java (we use kotlin)
* Style and grammar
* Some formatting
* Add links

I'll need you to:
* review the content is correct and accurate / suggest fixes.
* grammar review will be appreciated as well, as I'm not a native english speaker.

Just answer on the commit which way you want to go.

### CODE REVIEW

**Comment by nickbabcock** [comment]
Yes! This would be a nice improvement on the current docs. I read it and looks good, so once you finish up (kotlin -> java, etc), I'll go over it with a fine comb.

**Comment by baronlior** [comment]
OK. I'm good to go.

* please check the link on the last commit was placed properly - I wasn't able to verify that.

Cheers, Lior

**Comment by arteam** [comment]
Thank you!

File: docs/source/manual/internals.rst [hunk 1]
```diff
@@ -10,57 +10,143 @@ Congrats! Then you are ready to have a look into some nitty-gritty details of Dr
...
+*************
+
+Let us begin with a simplified version of the constructor:
+
+.. code-block:: java
+
+	public Cli(location : JarLocation, bootstrap : Bootstrap<?>, 
+			   stdOut: OutputStream, stdErr: OutputStream) {
+	  This.stdout = stdOut; this.stdErr = stdErr;
+	  val commands = new TreeMap<>();
```
Comment (jplock) [inline docs/source/manual/internals.rst]: some `val`'s snuck in here

**Pull request merged**